File: java/sfo_05_replace_spaces_s1/sfo_05_replace_spaces_s1.java
/*
* File: sfo_05_replace_spaces_s1.java
* Created Time: 2021-12-09
* Author: Krahets (krahets@163.com)
*/

package sfo_05_replace_spaces_s1;

import include.*;
import java.util.*;

// ===== Solution Code =====
class Solution {
    public String replaceSpace(String s) {
        StringBuilder res = new StringBuilder();
        for(Character c : s.toCharArray())
        {
            if(c == ' ') res.append("%20");
            else res.append(c);
        }
        return res.toString();
    }
}

public class sfo_05_replace_spaces_s1 {
    public static void main(String[] args) {
        // ======= Test Case =======
        String s = "We are happy.";
        // ====== Driver Code ======
        Solution slt = new Solution();
        String res = slt.replaceSpace(s);
        System.out.println(res);
    }
}

File: python/sfo_05_replace_spaces_s1.py
'''
File: sfo_05_replace_spaces_s1.py
Created Time: 2021-12-09
Author: Krahets (krahets@163.com)
'''

from include import *

# ===== Solution Code =====
class Solution:
    def replaceSpace(self, s: str) -> str:
        res = []
        for c in s:
            if c == ' ': res.append("%20")
            else: res.append(c)
        return "".join(res)

# ======= Test Case =======
s = "We are happy."
# ====== Driver Code ======
slt = Solution()
res = slt.replaceSpace(s)
print(res)
